server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -277,6 +277,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder.value(toString());
     }
 
+    public Version previousMajor() {
+        return Version.fromString(this.major - 1 + ".0.0");
```

Comment:
hm that would lead to a recursive creation. Maybe we could cache the string to build a previous version `major-1+."0.0"`  or a `byte (previousMajor-1)*1000000` 
then using Version.from* returns a cached instance